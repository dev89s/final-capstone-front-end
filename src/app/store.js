import { configureStore } from '@reduxjs/toolkit';

import dataReducer from '../components/Home/createSlice';
import userReducer from './user/userSlice';
import roomReducer from './room/roomSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    dataSlice: dataReducer,
    room: roomReducer,
  },
});

export default store;
