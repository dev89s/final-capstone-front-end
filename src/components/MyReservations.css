.page-container {
  background-image:
    linear-gradient(
      rgba(135, 206, 235, 0.5),
      rgba(135, 206, 235, 0.5)
    ),
    url('/public/pexels-sergio-souza-1782151.jpg'); /* Update with your image URL */

  background-size: cover;
  background-position: center;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  padding: 2rem 0;
}

.heading-container {
  background-color:
    rgba(
      255,
      255,
      255,
      0.8
    ); /* White color with transparency */

  padding: 10px 0;
  text-align: center;
  width: 100%;
}

.heading {
  margin: 0;
  font-size: 24px;
}

.reservations-container {
  margin: 20px;
  display: block;
  width: 80%;
  overflow-y: auto;
  max-height: 75vh;
  background-color: rgba(255, 255, 255, 0.8);
  border: 1px solid rgba(255, 255, 255, 0.8);
  padding-bottom: 0.8rem;
}

.reservation-list {
  width: 100%;
  border-collapse: collapse;
}

.reservation-item {
  background-color: rgba(255, 255, 255, 0.8);
  border: 1px solid #ccc;
  border-radius: 5px;
  margin: 10px 0;
}

.reservation-item td {
  padding: 10px;
  background-color: #e1eff1;
  text-align: center;
}

.reservation-item th {
  background-color: rgba(255, 255, 255, 0.8);
  padding: 10px;
  font-weight: bold;
}
