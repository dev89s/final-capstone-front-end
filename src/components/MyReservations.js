import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getUserReservations } from './api'; // Adjust the import path
import { setUserInfo } from '../app/user/userSlice';

import './MyReservations.css';

const MyReservations = () => {
  const userInfo = useSelector((state) => state.user.userInfo);
  const dispatch = useDispatch();

  const [reservations, setReservations] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    dispatch(setUserInfo(userInfo));
    const fetchReservations = async () => {
      try {
        const userReservations = await getUserReservations(userInfo.id);
        setReservations(userReservations);
        setLoading(false);
      } catch (error) {
        setError(error);
        setLoading(false);
      }
    };

    fetchReservations();
  }, [dispatch, userInfo.id]);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error fetching reservations.</div>;
  }

  return (
    <div className="page-container">
      <div className="heading-container">
        <h2 className="heading">My Reservations</h2>
      </div>
      <div className="reservations-container">
        <table className="reservation-list">
          <thead>
            <tr>
              <th>Room</th>
              <th>Date</th>
              <th>City</th>
            </tr>
          </thead>
          <tbody>
            {reservations.map((reservation) => (
              <tr key={reservation.id} className="reservation-item">
                <td>{reservation.room}</td>
                <td>{reservation.date}</td>
                <td>{reservation.city}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default MyReservations;
