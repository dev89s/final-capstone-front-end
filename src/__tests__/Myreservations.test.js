import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyReservations from '../components/MyReservations';
import store from '../app/store';

describe('Should render the reserver page', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MyReservations />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
