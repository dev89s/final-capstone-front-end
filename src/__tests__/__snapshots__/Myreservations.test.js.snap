// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the reserver page should render correctly 1`] = `
<div>
  Loading...
</div>
`;
